tests/utils/test_file.py

```diff
@@ -57,12 +57,24 @@ def test_fail_create_dir():
 
 def test_walk_tree():
     list_of_files = walk_tree(here, ".py")
+    print(f"Detected {len(list_of_files)} .py files found in directory: {here}")
     assert len(list_of_files) > 0
 
 
-@pytest.mark.xfail(
-    reason="This test is expected to fail since we passed invalid path for directory traversal",
-    raises=BuildTestError,
-)
-def test_walk_tree_invalid_dir():
-    walk_tree("/xyz", ".py")
+def test_walk_tree_no_files(tmp_path):
+    list_of_files = walk_tree(tmp_path, ".py")
+    print(f"Detected {len(list_of_files)} .py files found in directory: {tmp_path}")
+    assert 0 == len(list_of_files)
```

Comment:
This test could be improved to actually test that it's walking a directory (e.g., os.listdir()) would have worked here, there is no tree walking. But this is a nit, and otherwise the PR looks good.